pkg/chunk/purger/purger.go

```diff
@@ -424,6 +472,9 @@ func (dp *DataPurger) pullDeleteRequestsToPlanDeletes() error {
 		dp.executePlansChan <- req
 	}
 
+	dp.metrics.oldestPendingDeleteRequestAgeSeconds.Set(float64(now.Sub(oldestPendingRequestCreatedAt) / time.Second))
+	dp.metrics.pendingDeleteRequestsCount.Set(float64(pendingDeleteRequestsCount))
+
```

Comment:
This will only update metrics after all delete requests have plans for them built. Is that OK?

Reply:
Yes, that is the only option right now because if we change the count as we go through each delete request the partially updated value could get scraped and we would see unexpected jumps in the number of pending requests.

Reply:
This could be done by refactoring the code to find the requests to process as well as do a count of pending requests and build the plans after updating the metric. What do you think?